app/classes/content_filter.rb

```diff
@@ -1,96 +1,117 @@
+  # an unnecessary step for new observation filter -- it's done
+  # automatically.)
+  #
+  #   Add tests, e.g., to TestUserContentFilter
+  #   Supplement fixtures as needed by added tests
+  #   Add a filter definition below
+  # * Add a new checkbox_val method in User
+  #   In /config/locales/en.txt define text to be displayed before and next to
+  #     check boxes, using prefs_obs_filters_has_images: as a model.
+  # * Supplement _prefs_filters.html.erb as needed
+  # * Add filter to AccountController#prefs_types & #update_content_filter
+  #   For Observation filter, supplement Query::Initializers::ObservationFilters
+  # * Supplement ApplicationController#show_index_of_objects as needed
+  #   To filter another object, create a new initializer and include in
+  #     appropriate searches.
+  #
+  # To be able to override the new filter in Advanced Searches, at least:
+  #   Add tests, e.g., to TestAdvancedSearchFilters
+  #   Supplement fixtures as needed by added tests
+  #   In /config/locales/en.txt define text to be displayed before and next to
+  #     radio boxes, using advanced_search_filter_has_images: as a model.
+  # * Supplement _advanced_search_filters.html.erb as needed
+  # * Supplement ObservationController#advanced_search_form as needed.
+  # * Supplement ApplicationController#show_index_of_objects as needed.
+  # * Supplement Query::RssLogBase as needed.
+  #
+  # To also add it as a Pattern Search
+  #   Supplement PatternSearchTest
+  #   Supplement classes/pattern_search/observation.rb or -- if it's not an
+  #     Observation search -- add a new pattern_search file and class.
+  #
+  # There are probably other steps/files I've forgotten. JDC 2016-09-15
 
-  def has_images
-    {
-      sym:          :has_images,
-      model:        Observation,
-      checkbox:     :has_images_checkbox,
-      checked_val:  "NOT NULL",
-      off_val:      "off",
-      on_vals:      ["NOT NULL", "NULL"],
-      sql_cond:     "observations.thumb_image_id IS #{params[:has_images]}"
-    }
-  end
+  attr_accessor :sym
+  attr_accessor :model
+  attr_accessor :checkbox
+  attr_accessor :on_vals
+  attr_accessor :checked_val
+  attr_accessor :off_val
 
-  def has_specimen
-    {
-      sym:          :has_specimen,
-      model:        Observation,
-      checkbox:     :has_specimen_checkbox,
-      checked_val:  "TRUE",
-      off_val:      "off",
-      on_vals:      ["TRUE", "FALSE"],
-      sql_cond:     "observations.specimen IS #{params[:has_specimen]}"
-   }
+  def initialize(opts)
+    @sym          = opts[:sym]
+    @model        = opts[:model]
+    @checkbox     = opts[:checkbox]
+    @on_vals      = opts[:on_vals]
+    @checked_val  = opts[:checked_val]
+    @off_val      = opts[:off_val]
   end
 
-  def filters
-    [has_images, has_specimen]
+  # Array of all available filters.
+  def self.all
+    @@filters ||= [
+      ContentFilter.new(
+        sym:          :has_images,
+        model:        Observation,
```

Comment:
Pasting @pellaea comments here, to keep thread together:
> re: observations only -- See use case in other email. Filtering out
lichens should apply to names as well as observations.

and for locations. If I'm interested in Observations only if they're from North America, I probably don't want my RSS feed to include new Locations in Laos.

But am I guilty of premature generalization?